test/emu_mps/test_mps_config.py

```
@@ -34,3 +50,124 @@ def test_unsupported_noise():
+    default_config = EmulationConfig(
+        dt=dt,
+        precision=precision,
+        max_bond_dim=max_bond_dim,
+        max_krylov_dim=max_krylov_dim,
+        extra_krylov_tolerance=extra_krylov_tolerance,
+        num_gpus_to_use=num_gpus_to_use,
+        observables=observables,
+    )
+
+    config_str = default_config.to_abstract_repr()
+    deserialized_config = MPSConfig.from_abstract_repr(config_str)
+
+    values = [
+        dt,
+        precision,
+        max_bond_dim,
+        max_krylov_dim,
+        extra_krylov_tolerance,
+        num_gpus_to_use,
+        observables,
+    ]
+
+    for attr, val in zip(mps_attributes, values):
+        assert getattr(deserialized_config, attr) == getattr(
+            default_config, attr
+        ), f"{attr} mismatch"
+        assert getattr(deserialized_config, attr) == val, f"{attr} != {val} mismatch"
+
+
+def test_default_constructors_for_all_config() -> None:
+    # BackendConfig
+    msg = (
+        "'BackendConfig' received unexpected keyword arguments: "
+        "{'blabla'}; only the following keyword arguments "
+        "are expected: set(). "
+    )
+    # ^ and $ are for full regex match
+    # re.escape() avoid interpreting message symbols "{}"", "()","."" as regex
+    with pytest.warns(UserWarning, match="^" + re.escape(msg) + "$"):
+        BackendConfig(blabla=10)
+
+    # EmulationConfig No warning
+    with warnings.catch_warnings(record=True) as w:
+        warnings.simplefilter("always")
+        EmulationConfig(blabla=10, observables=[BitStrings(evaluation_times=[1.0])])
+        assert not w, "Unexpected warnings: No warning"
+
+    # MPSConfig
+    msg = (
+        "'MPSConfig' received unexpected keyword arguments: "
+        "{'blabla'}; only the following keyword arguments "
+        "are expected:"
+    )
+    # re.escape() avoid interpreting message symbols "{}"", "()","."" as regex
+    with pytest.warns(UserWarning, match="^" + re.escape(msg)):
+        MPSConfig(blabla=10)
```

- `BackendConfig` should not be tested here
- `EmulationConfig`: rather than not catching a warning I would just test the expected behavior. This is tested in pulser so I would remove this

The third one is a test on the expected_kwargs patch.
I would call it `test_expected_kwargs`.
Overall I would just replace all with:
```
def test_expected_kwargs() -> None:
    with pytest.warns(UserWarning, match="received unexpected keyword arguments"):
        MPSConfig(blabla=10)
```

---

On top of that, I would prefer an error rather than a warning

---

The test is designed to demonstrate for users/devs that there are 3 different behaviours.

I do it exactly to capture that `MPS/SVConfig` **does** verify kwargs, while `EmulationConfig` does **not**. It is inexplicit knowledge. 
Before PR we didn't notice that behaviour was changed. 

Though I don't strongly insist to have all 3 here, I think they are complementary demonstrative

---

changing it to error requires design work on pulser side. Or some bigger work on our side.

---

one option could be
```
        if invalid_kwargs := set(kwargs) - super()._expected_kwargs():
            raise TypeError(
                f"MPSConfig received unexpected keyword arguments: "
                f"{invalid_kwargs}; only the following keyword "
                f"arguments are expected: {self._expected_kwargs()}. "
            )
```

because kwargs should only be valid keywords for EmulationConfig

---

Testing `EmulationConfig` causes the tests to no longer be "unit tests" in the sense that they start testing things outside of this repo. I would just test `MPS/SVConfig` in our tests, and leave responsibility for `EmulationConfig` to pulser.

---

Ok, so I convinced Henrique that we should just raise an error on the pulser side. So let's leave it at a warning for now, and then change when that gets released in pulser. I still think we should not test `BackendConfig` and `EmulationConfig` in this repo though, that is pulser's responsibility.